lib/flatpak-installation.c

```diff
@@ -1121,6 +1121,31 @@ get_write_progress (guint outstanding_writes)
   return (guint) (3 / (gdouble) outstanding_writes);
 }
 
+static gboolean
+app_contains_extra_data (const char *app_id)
+{
+  /* HACK: This is a temporary solution until the detached metadata work is ready */
+  const char *extra_data_apps[] =
```

Comment:
This is just the way flatpak handles it. You can do a metadata only pull and examine it. We do that in eos-updater, but more because it predated the summary file. Speaking of which, we can always have flatpak include this information in the summary file, which I'm sure gnome-software is already relying on.

Not saying this won't work as is, but there are certainly ways to get this information.